### PR TITLE
fix search

### LINKED ISSUE #666: search failures on website
Any search on https://wechaty.js.org/ will result in _"Page Not Found"_ error --

`We could not find what you were looking for.`

As per https://github.com/wechaty/wechaty.js.org/issues/440#issuecomment-778718023

> @suntong Yes, please file a new issue for search failures.
> 
> We have some problem in the search part of our website, and we need to fix it.

### PR DESCRIPTION
**Pull request by l3ob**
fix https://github.com/wechaty/wechaty.js.org/issues/666

### CODE REVIEW

**Comment by l3ob** [comment]
# 0x00
search result of jekyll posts cannot route correctly in docusaurus project
so change history.push(URL) to location.href=URL

script to copy docusaurus theme:
```
npx docusaurus swizzle @docusaurus/theme-search-algolia SearchBar --danger
```

# 0x01
remove some params to show all results

script to copy docusaurus theme:
```
npx docusaurus swizzle @docusaurus/theme-search-algolia SearchPage --danger
```

**Review by huan — CHANGES_REQUESTED** [review]
It's great to see this PR for fixing the search function of our site!

Could you please sign the CLA first, so that we can be able to merge this PR?

And also, I'd like to know what exactly diff did you make for this modification? Could you please submit a diff/patch so that we can better understand this problem, and be able to review it later?

Thank you very much!

**Comment by l3ob** [comment]
@huan the CLA is signed, but I don't know why the status not changed yet.

image

**Comment by l3ob** [comment]
> It's great to see this PR for fixing the search function of our site!
> 
> Could you please sign the CLA first, so that we can be able to merge this PR?
> 
> And also, I'd like to know what exactly diff did you make for this modification? Could you please submit a diff/patch so that we can better understand this problem, and be able to review it later?
> 
> Thank you very much!

https://github.com/l3ob/wechaty.js.org/commit/cbb6ef222fee818e3ff0d0ac9c9f4bfb8dd41b07

I start a branch here to better explain the change.

First, jekyll posts are not belongs to docusaurus project, so the default method using docusaurus router cannot find the posts. Redirect search result with href instead of router/push fix the problem.

Follow docusaurus docs to copy and edit the component:
https://docusaurus.io/docs/search#editing-the-algolia-search-component

Second, the search page show no results. I try to modify the algolia http request and removing facetFilters params could do the trick. Guess it's something with the algolia configuration. For now we can just comment the lines.

**Comment by huan** [comment]
> @huan the CLA is signed, but I don't know why the status not changed yet.

According to the comment replied by the CLA bot, i3ob is OK but there have commits made by ❌ coderprepares.

You can pass the CLA check either by removing the commits from coderprepares or signing the CLA with coderprepares.

**Comment by l3ob** [comment]
> > @huan the CLA is signed, but I don't know why the status not changed yet.
> 
> According to the comment replied by the CLA bot, i3ob is OK but there have commits made by ❌ coderprepares.
> 
> You can pass the CLA check either by removing the commits from coderprepares or signing the CLA with coderprepares.

It's my another account. CLA sign is done. What else can I do?

File: docusaurus/src/theme/SearchBar/index.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+import Head from '@docusaurus/Head';
+import useSearchQuery from '@theme/hooks/useSearchQuery';
+import {DocSearchButton, useDocSearchKeyboardEvents} from '@docsearch/react';
+import useAlgoliaContextualFacetFilters from '@theme/hooks/useAlgoliaContextualFacetFilters';
+import {translate} from '@docusaurus/Translate';
+
+let DocSearchModal = null;
+
+function Hit({hit, children}) {
+  return <Link href={hit.url}>{children}</Link>;
```
Comment (l3ob) [inline docusaurus/src/theme/SearchBar/index.js:23]: change `<Link to={hit.url}>` to `<Link href={hit.url}>`

File: docusaurus/src/theme/SearchBar/index.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+        setInitialQuery(event.key);
+      });
+    },
+    [importDocSearchModalIfNeeded, setIsOpen, setInitialQuery],
+  );
+
+  const navigator = useRef({
+    navigate({itemUrl}) {
+      // history.push(itemUrl);
+      location.href = itemUrl;
```
Comment (l3ob) [inline docusaurus/src/theme/SearchBar/index.js:105]: change `history.push(itemUrl);` to `location.href = itemUrl;`

File: docusaurus/src/theme/SearchPage/index.js [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+
+    // Object.entries(docsSearchVersionsHelpers.searchVersions).forEach(
+    //   ([pluginId, searchVersion]) => {
+    //     algoliaHelper.addDisjunctiveFacetRefinement(
+    //       'docusaurus_tag',
+    //       `docs-${pluginId}-${searchVersion}`,
+    //     );
+    //   },
+    // );
+
```
Comment (l3ob) [inline docusaurus/src/theme/SearchPage/index.js:286]: remove certain search params

File: docusaurus/src/theme/SearchPage/index.js [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+    // algoliaHelper.addDisjunctiveFacetRefinement('language', currentLocale);
+
+    // Object.entries(docsSearchVersionsHelpers.searchVersions).forEach(
+    //   ([pluginId, searchVersion]) => {
+    //     algoliaHelper.addDisjunctiveFacetRefinement(
+    //       'docusaurus_tag',
+    //       `docs-${pluginId}-${searchVersion}`,
+    //     );
+    //   },
+    // );
```
Comment (l3ob) [inline docusaurus/src/theme/SearchPage/index.js:285]: remove certain search params

**Comment by huan** [comment]
Merge to #902

**Pull request closed without merging**